### ISSUE TITLE
custom errorMessage by field name doesn't work.

### ISSUE THREAD

**Issue by sandrina-p** [remoteoss/json-schema-form#55]
Given the following JSON Schema, I cannot override its `errorMessage` using `createHeadlessForm()`

```json
{
  "properties": {
    "year": {
      "title": "Year",
      "type": "number",
      "presentation": {
        "inputType": "number"
      },
      "x-jsf-errorMessage": {
        "required": "Cannot be empty."
      },
      "minimum": 1900,
      "maximum": 2023
    }
  },
  "required": ["year"]
}
```

JSF Config:

```js
const { fields } = createHeadlessForm(schemaAbove, {
  customProperties: {
    year: {
      errorMessage: {
        required: "The year is mandatory."
      }
    }
  }
})
```

**Comment by sandrina-p** [comment]
This issue might be out of scope for `json-schema-form`. We are discussing internally the possibility of [deprecating `customProperties`](https://linear.app/remote/issue/RMT-696/jsf-config-customproperties-deprecation) in favor of moving mutations to JSON Schemas outside `createHeadlessForm()`.

So in this case, we'd instead recommend doing this:

```js
import merge from 'lodash/fp/merge';

const newSchema = merge(schema, {
  properties: {
    year: {
      "x-jsf-errorMessage": {
        required: "Cannot be empty (custom error)"
      }
    },
  }
});

const form = createHeadlessForm(newSchema)

```

**Comment by sandrina-p** [comment]
This is solved in favor of the new `modify()` module! #75
